app/Http/Controllers/Admin/AdminCheckoutController.php

```diff
@@ -0,0 +1,126 @@
+                ->get();
+            $transactions['expense'] = $semester->transactionsInCheckout($checkout)
+                ->where('payment_type_id', PaymentType::expense()->id)
+                ->get();
+            $transactions['netreg'] = $checkout->netregSum($semester);
+            $transactions['print'] = $checkout->printSum($semester);
+            $transactions['sum'] = $semester->transactionsInCheckout($checkout)
+                ->sum('amount');
+
+            $data[$semester->tag().' ('.$semester->getStartDate()->format('Y.m.d').'-'.$semester->getEndDate()->format('Y.m.d').')'] = [
+                'transactions' => $transactions,
+                'workshop_balances' => $semester->workshopBalances,
+            ];
+        }
+
+        $current_balance = $checkout->balance();
+        $current_balance_in_checkout = $checkout->balanceInCheckout();
+
+        return view('admin.checkout', [
+            'data' => $data,
+            'my_transactions' => $my_transactions_not_in_checkout,
+            'sum_my_transactions' => $sum,
+            'current_balance' => $current_balance,
+            'current_balance_in_checkout' => $current_balance_in_checkout, ]);
+    }
+
+    public function printToCheckout(Request $request)
+    {
+        $checkout = Checkout::admin();
+        $this->authorize('addPayment', $checkout);
+
+        /* Moving the Netreg amount from Valasztmany to Admins is not tracked (yet) */
+        $validator = Validator::make($request->all(), [
+            'password' => 'required|in:'.$checkout->password,
+        ]);
+        $validator->validate();
+        if ($validator->fails()) {
+            return back()->withErros($validator)->withInput();
+        }
+
+        $transactions = Transaction::where('receiver_id', Auth::user()->id)
+            ->where('moved_to_checkout', null)
+            ->where('payment_type_id', PaymentType::print()->id)
+            ->get();
+        foreach ($transactions as $transaction) {
+            $transaction->update([
+                'moved_to_checkout' => Carbon::now(),
+            ]);
+        }
+
+        return redirect()->back()->with('message', __('general.successfully_added'));
+    }
+
+    public function addTransaction(Request $request)
+    {
+        $checkout = Checkout::admin();
+        $this->authorize('administrate', $checkout);
+
+        $validator = Validator::make($request->all(), [
+            'comment' => 'required|string',
+            'amount' => 'required|integer',
+            'password' => 'required|in:'.$checkout->password,
+        ]);
+        $validator->validate();
+        if ($validator->fails()) {
+            return back()->withErros($validator)->withInput();
+        }
+
+        $type = $request->amount > 0 ? PaymentType::income()->id : PaymentType::expense()->id;
+
+        // $this->createTransaction($checkout, null, Auth::user()->id, $request->amount, $type, Carbon::now(), $request->comment);
+
+        Transaction::create([
```

Comment:
Please use the commented function from above.

Reply:
I think we should use functions with named parameters in cases where there are lots of parameters and default values. PHP 8 adds this functionality. If we can't use that, I'm for using the `Transaction::create()` function, as one can see the names of the parameters.

Reply:
Can't we write `someFun($name = $value)`?
If not, the convention is commenting the parameter where it's not obvious, like that: `someFun(/* receiver */ null)`.

Reply:
I get your idea, but I really hate code duplications and for me it's a too simple code duplicated at too many places.
I will let @kdmnk decide.

Reply:
I found this: https://stitcher.io/blog/php-8-named-arguments

That was the thing why I used the `Transaction::create()`, because it was defined in the `EconomicController `and I didn't want do define it again in the AdminCheckoutController. Also I found the original approach more intuitive.

Reply:
Hm, yeah, you probably right.

Reply:
Okay, let's go with it. Would it be a big ask to change the ones I replaced with this functions?